pkg/utils/release/controller.go

```diff
@@ -394,3 +400,119 @@ func (s *SuiteController) GetSbomPyxisByImageID(pyxisStageURL, imageID string,
+				if errors.IsNotFound(err) {
+					// napshotEnvironmentBinding CR is already removed
+					return true, nil
+				}
+				g.GinkgoWriter.Printf("unable to retrieve napshotEnvironmentBinding '%s' in '%s': %v\n", snapshotEnvironmentBindingCR.Name, snapshotEnvironmentBindingCR.Namespace, err)
+				return false, nil
+
+			}
+
+			// Remove the finalizer, so that it can be deleted.
+			snapshotEnvironmentBindingCR.Finalizers = []string{}
+			if err := s.KubeRest().Update(context.TODO(), &snapshotEnvironmentBindingCR); err != nil {
+				g.GinkgoWriter.Printf("unable to remove finalizers from snapshotEnvironmentBinding '%s' in '%s': %v\n", snapshotEnvironmentBindingCR.Name, snapshotEnvironmentBindingCR.Namespace, err)
+				return false, nil
+			}
+
+			if err := s.KubeRest().Delete(context.TODO(), &snapshotEnvironmentBindingCR); err != nil {
+				g.GinkgoWriter.Printf("unable to delete napshotEnvironmentBinding '%s' in '%s': %v\n", snapshotEnvironmentBindingCR.Name, snapshotEnvironmentBindingCR.Namespace, err)
+				return false, nil
+			}
+			return true, nil
+		})
+		if err != nil {
+			return fmt.Errorf("deletion of napshotEnvironmentBinding '%s' in '%s' timed out", snapshotEnvironmentBinding.Name, snapshotEnvironmentBinding.Namespace)
+		}
+
+	}
+
+	return nil
+}
+
+func (s *SuiteController) WaitForReleasePipelineToBeFinished(namespace string, maxRetries int) error {
+	attempts := 1
+	var pr *v1beta1.PipelineRun
+
+	for {
+		err := wait.PollImmediate(20*time.Second, 30*time.Minute, func() (done bool, err error) {
+
+			prList, err := s.GetAllReleasePipelineRunInNamespace(namespace)
+			if err != nil {
+				g.GinkgoWriter.Println("PipelineRun has not been created yet")
+				return false, nil
+			}
+
+			pr := &prList.Items[0]
+			g.GinkgoWriter.Printf("PipelineRun %s reason: %s\n", pr.Name, pr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).GetReason())
+
+			if !pr.IsDone() {
+				return false, nil
+			}
+
+			if pr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsTrue() {
+				return true, nil
+			} else {
+				var prLogs string
+				if err = tekton.StorePipelineRun(pr, s.KubeRest(), s.KubeInterface()); err != nil {
+					g.GinkgoWriter.Printf("failed to store PipelineRun %s:%s: %s\n", pr.GetNamespace(), pr.GetName(), err.Error())
+				}
+				if prLogs, err = tekton.GetFailedPipelineRunLogs(s.KubeRest(), s.KubeInterface(), pr); err != nil {
+					g.GinkgoWriter.Printf("failed to get logs for PipelineRun %s:%s: %s\n", pr.GetNamespace(), pr.GetName(), err.Error())
+				}
+				return false, fmt.Errorf(prLogs)
+			}
+		})
+
+		if err != nil {
+			g.GinkgoWriter.Printf("attempt %d/%d: PipelineRun %q failed: %+v", attempts, maxRetries+1, pr.GetName(), err)
+			// Retry the PipelineRun only in case we hit the known issue https://issues.redhat.com/browse/SRVKP-2749
+			if attempts == maxRetries+1 || pr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).GetReason() != "CouldntGetTask" {
+				return err
+			}
+			attempts++
```

Comment:
I don't think the retry logic makes sense for release pipelineruns (at least not with this approach).

What it'd currently do is that in case of `CouldntGetTask` error it'd call the `wait.PollImmediate` func `maxRetries` times, but on the same release pipelinerun, which does not help.

I'd keep `WaitForReleasePipelineToBeFinished` method, but remove the `retry` part of the code.

Reply:
you are right, my bad 
I forget to remove it